backend/gradle.lockfile

```diff
@@ -0,0 +1,206 @@
+# This is a Gradle generated file for dependency locking.
```

Comment:
generated with `./gradlew dependencies --write-locks`

Reply:
Is this new, and would we need to do this again periodically? Wondering if we should add something about it to the wiki.

Reply:
No its nothing new, whats new is the new gradle version now combines the [compileClasspath.lockfile](https://github.com/CDCgov/prime-simplereport/pull/4243/files#diff-21c40d80236ff73bc3b206e7d71cecd5af9e9a6c7aac467e45b05763d05be6d4), and  [runtimeClasspath.lockfile](https://github.com/CDCgov/prime-simplereport/pull/4243/files#diff-c45bf0535ff62b9b68475784703607121c6042c387804a1648c333e991626677) files into one file.
here are the docs, https://docs.gradle.org/current/userguide/dependency_locking.html
we generally have to re-generate this file everytime we update the gradle dependencies. 
Its analgous to `npm lockfile`, it wouldn't hurt to have a wiki about it, though its part of knowing the framework and its docs, so I can go both ways.

Reply:
thanks for the explanation about the combination! if that's the only difference, I agree we probably don't need to document it. I will say I've been on a few projects that don't use gradle dependency locking, and didn't know this was best practice so ty 😬